specification/eventgrid/data-plane/Microsoft.EventGrid/2018-01-01/EventGrid.json

```diff
@@ -0,0 +1,127 @@
+    ],
+    "paths": {
+        "/api/events": {
+            "post": {
+                "tags": [
+                    "Events"
+                ],
+                "description": "Publishes a batch of events to an Azure Event Grid topic.",
+                "operationId": "PublishEvents",
+                "parameters": [
+                    {
+                        "name": "events",
+                        "description": "An array of events to be published to Event Grid.",
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/EventGridEvent"
+                            }
+                        }
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK"
+                    },
+                    "default": {
+                        "description": "*** Error Responses: ***\n\n * 400 Bad Request.\n\n * 500 Internal Server Error.\n\n * 429 Too Many Events\n\n * 404 Not Found."
+                    }
+                }
+            }
+        }
+    },
+    "definitions": {
+        "EventGridEvent": {
+            "type": "object",
+            "description": "Properties of an event published to an Event Grid topic.",
+            "required": [
+                "id",
+                "subject",
+                "data",
+                "eventType",
+                "eventTime",
+                "dataVersion"
+            ],
+            "properties": {
+                "id": {
+                    "description": "An unique identifier for the event.",
+                    "type": "string"
+                },
+                "topic": {
+                    "description": "The resource path of the event source.",
+                    "type": "string"
+                },
+                "subject": {
+                    "description": "A resource path relative to the topic path.",
+                    "type": "string"
+                },
+                "data": {
+                    "description": "Event data specific to the event type.",
+                    "type": "object"
+                },
+                "eventType": {
+                    "description": "The type of the event that occurred.",
+                    "type": "string"
+                },
+                "eventTime": {
+                    "description": "The time (in UTC) the event was generated.",
+                    "format": "date-time",
+                    "type": "string"
+                },
+                "metadataVersion": {
+                    "description": "The schema version of the event metadata.",
+                    "readonly": true,
```

Comment:
How does the value for this property get set?  Usually fields marked as read-only are immutable values returned from the API.

Reply:
The value for this property gets set by the Event Grid service. There are two use cases of this EventGridEvent definition: 1) Publishers publishing events to EventGrid 2) Consumers consuming events delivered by EventGrid.  The reason this property is in this definition is because of 2) above where consuming applications may want to know this version.

Reply:
I think the casing here is incorrect which is what's tripping up the validation tools, it should be ```readOnly```.